runtime/reconcilers/metrics_view.go

```diff
@@ -152,6 +153,27 @@ func (r *MetricsViewReconciler) validate(ctx context.Context, mv *runtimev1.Metr
 	return errors.Join(errs...)
 }
 
+func validateDimension(ctx context.Context, olap drivers.OLAPStore, t *drivers.Table, d *runtimev1.MetricsViewSpec_DimensionV2, fields map[string]*runtimev1.StructType_Field) error {
+	if d.Expression == "" {
+		if d.Column == "" {
+			return nil
+		}
+		if _, isColumn := fields[d.Column]; !isColumn {
+			return fmt.Errorf("failed to validate dimension %q: column %q not found in table", d.Name, d.Column)
+		}
+		return nil
+	}
+
+	err := olap.Exec(ctx, &drivers.Statement{
+		Query:  fmt.Sprintf("SELECT %s FROM %s GROUP BY 1", d.Expression, safeSQLName(t.Name)),
```

Comment:
Should we put parentheses around the expression, i.e. `SELECT (%s) ...`? In case the expression is a sub-query

Reply:
Might this also be needed elsewhere?

Reply:
Yes adding them